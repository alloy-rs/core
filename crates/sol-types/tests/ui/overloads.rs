use alloy_sol_types::sol;

mod function {
    use super::*;

    sol! {
        function overloaded();
        function overloaded(uint256);
        function overloaded(uint256,address);
        function overloaded(address);
        function overloaded(address,string);
    }

    sol! {
        function overloadTaken();
        function overloadTaken(uint256);
        function overloadTaken_0();
        function overloadTaken_1();
        function overloadTaken_2();
    }

    sol! {
        function sameOverload();
        function sameOverload();
    }

    sol! {
        function sameTysOverload1(uint256[]memory a);
        function sameTysOverload1(uint256[]storage b);
    }

    sol! {
        function sameTysOverload2(string memory,string storage);
        function sameTysOverload2(string storage b,string calldata);
    }
}

mod event {
    use super::*;

    sol! {
        event overloaded();
        event overloaded(uint256);
        event overloaded(uint256,address);
        event overloaded(address);
        event overloaded(address,string);
    }

    sol! {
        event overloadTaken();
        event overloadTaken(uint256);
        event overloadTaken_0();
        event overloadTaken_1();
        event overloadTaken_2();
    }

    sol! {
        event sameOverload();
        event sameOverload();
    }

    sol! {
        event sameTysOverload1(uint256[] a);
        event sameTysOverload1(uint256[] b);
    }

    sol! {
        event sameTysOverload2(string, string);
        event sameTysOverload2(string, string);
    }
}

/*
mod error {
    use super::*;

    sol! {
        error overloaded();
        error overloaded(uint256);
        error overloaded(uint256,address);
        error overloaded(address);
        error overloaded(address,string);
    }

    sol! {
        error overloadTaken();
        error overloadTaken(uint256);
        error overloadTaken_0();
        error overloadTaken_1();
        error overloadTaken_2();
    }

    sol! {
        error sameOverload();
        error sameOverload();
    }

    sol! {
        error sameTysOverload1(uint256[] a);
        error sameTysOverload1(uint256[] b);
    }

    sol! {
        error sameTysOverload2(string, string);
        error sameTysOverload2(string, string);
    }
}
*/

fn main() {}
