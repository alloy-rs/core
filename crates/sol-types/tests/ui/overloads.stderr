error: function `overloadTaken` is overloaded, but the generated name `overloadTaken_0` is already in use
  --> tests/ui/overloads.rs:15:18
   |
15 |         function overloadTaken();
   |                  ^^^^^^^^^^^^^
   |
note: other declaration is here
  --> tests/ui/overloads.rs:17:18
   |
17 |         function overloadTaken_0();
   |                  ^^^^^^^^^^^^^^^

error: function `overloadTaken` is overloaded, but the generated name `overloadTaken_1` is already in use
  --> tests/ui/overloads.rs:16:18
   |
16 |         function overloadTaken(uint256);
   |                  ^^^^^^^^^^^^^
   |
note: other declaration is here
  --> tests/ui/overloads.rs:18:18
   |
18 |         function overloadTaken_1();
   |                  ^^^^^^^^^^^^^^^

error: function with same name and parameter types defined twice
  --> tests/ui/overloads.rs:23:18
   |
23 |         function sameOverload();
   |                  ^^^^^^^^^^^^
   |
note: other declaration is here
  --> tests/ui/overloads.rs:24:18
   |
24 |         function sameOverload();
   |                  ^^^^^^^^^^^^

error: function with same name and parameter types defined twice
  --> tests/ui/overloads.rs:28:18
   |
28 |         function sameTysOverload1(uint256[]memory a);
   |                  ^^^^^^^^^^^^^^^^
   |
note: other declaration is here
  --> tests/ui/overloads.rs:29:18
   |
29 |         function sameTysOverload1(uint256[]storage b);
   |                  ^^^^^^^^^^^^^^^^

error: function with same name and parameter types defined twice
  --> tests/ui/overloads.rs:33:18
   |
33 |         function sameTysOverload2(string memory,string storage);
   |                  ^^^^^^^^^^^^^^^^
   |
note: other declaration is here
  --> tests/ui/overloads.rs:34:18
   |
34 |         function sameTysOverload2(string storage b,string calldata);
   |                  ^^^^^^^^^^^^^^^^

error: event `overloadTaken` is overloaded, but the generated name `overloadTaken_0` is already in use
  --> tests/ui/overloads.rs:50:15
   |
50 |         event overloadTaken();
   |               ^^^^^^^^^^^^^
   |
note: other declaration is here
  --> tests/ui/overloads.rs:52:15
   |
52 |         event overloadTaken_0();
   |               ^^^^^^^^^^^^^^^

error: event `overloadTaken` is overloaded, but the generated name `overloadTaken_1` is already in use
  --> tests/ui/overloads.rs:51:15
   |
51 |         event overloadTaken(uint256);
   |               ^^^^^^^^^^^^^
   |
note: other declaration is here
  --> tests/ui/overloads.rs:53:15
   |
53 |         event overloadTaken_1();
   |               ^^^^^^^^^^^^^^^

error: event with same name and parameter types defined twice
  --> tests/ui/overloads.rs:58:15
   |
58 |         event sameOverload();
   |               ^^^^^^^^^^^^
   |
note: other declaration is here
  --> tests/ui/overloads.rs:59:15
   |
59 |         event sameOverload();
   |               ^^^^^^^^^^^^

error: event with same name and parameter types defined twice
  --> tests/ui/overloads.rs:63:15
   |
63 |         event sameTysOverload1(uint256[] a);
   |               ^^^^^^^^^^^^^^^^
   |
note: other declaration is here
  --> tests/ui/overloads.rs:64:15
   |
64 |         event sameTysOverload1(uint256[] b);
   |               ^^^^^^^^^^^^^^^^

error: event with same name and parameter types defined twice
  --> tests/ui/overloads.rs:68:15
   |
68 |         event sameTysOverload2(string, string);
   |               ^^^^^^^^^^^^^^^^
   |
note: other declaration is here
  --> tests/ui/overloads.rs:69:15
   |
69 |         event sameTysOverload2(string, string);
   |               ^^^^^^^^^^^^^^^^
